File: 59 Spiral Matrix II.cpp
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        int num = 1;
        vector<vector<int>> matrix(n, vector<int>(n, 0));
        for(int cnt = 0; cnt < n - 1; cnt++) {
            for (int i = cnt; i < n - cnt - 1; i++)
                matrix[cnt][i] = num++;
            for (int j = cnt; j < n - cnt - 1; j++)
                matrix[j][n - 1 - cnt] = num++;
            for (int x = n - cnt - 1; x > cnt; x--)
                matrix[n - 1 - cnt][x] = num++;
            for (int y = n - cnt - 1; y > cnt; y--)
                matrix[y][cnt] = num++;
        }
        // 奇数时，需要单独赋值
        if (n % 2) {
            matrix[n/2][n/2] = num;
        }
        return matrix;
    }
};

File: 59 Spiral Matrix II.py
def generateMatrix(n: int):
    nums = [[0] * n for _ in range(n)]
    loop = n // 2 
    cnt = 1
    for i in range(loop):
        for j in range(i, n - 1 - i):
            nums[i][j] = cnt
            cnt += 1
        for j in range(i, n - 1 - i):
            nums[j][n-i-1] = cnt
            cnt += 1
            
        # tmp = [temp for temp in range(n-i-1, i, -1)]
        # print(f"tmp: {tmp}")
        # print(range(i+1, n-i, -1))
        # print(range(n-i-1, i, -1))
        for j in range(n-i-1, i, -1):
            print(f"n-i-1: {n-i-1}, j: {j}")
            nums[n-i-1][j] = cnt
            cnt +=1
        for j in range(n-i-1, i, -1):
            print(f"j: {j}, i: {i}")
            nums[j][i] = cnt
            cnt +=1
    print(n//2)
    if n % 2 == 1:
        nums[n//2][n//2] = cnt
    return nums

n = 3
print(generateMatrix(n))